bin/rbenv-sh-update

```diff
@@ -4,33 +4,37 @@ set -e
 
 shell="$(basename "${RBENV_SHELL:-$SHELL}")"
 
-case "$shell" in
-fish )
-  cat <<EOF
-command rbenv update $@
-if command test -t 1
-  printf "\\e[1;32mreloading rbenv\\e[0m\\n"
-  . (rbenv init -|psub)
-  printf " \\033[1;32m|\\033[0m  done\\n"
-else
-  printf "reloading rbenv\\n"
-  . (rbenv init -|psub)
-  printf " |  done\\n"
-end
-EOF
-  ;;
-* )
-  cat <<EOF
-command rbenv update $@;
-if [ -t 1 ]; then
-  printf "\\e[1;32mreloading rbenv\\e[0m\\n";
-  eval "\$(rbenv init -)";
-  printf " \\033[1;32m|\\033[0m  done\\n";
-else
-  printf "reloading rbenv\\n";
-  eval "\$(rbenv init -)";
-  printf " |  done\\n";
-fi
-EOF
-  ;;
+# Parse command line flags
+case "$1" in
+  -n | --noop ) noop=true ;;
+  -q | --quiet | --silent ) quiet=true ;;
 esac
+
+
+
+echo "if command rbenv update $@; then"
+
+if [ -z "$quiet" ]; then
+  echo "  if command test -t 1; then"
+  echo "    printf \"\\e[1;32mReloading rbenv\\e[0m\\n\";"
+  echo "  else"
+  echo "    printf \"Reloading rbenv\\n\";"
+  echo "  fi;"
+fi
+
+if [ -z "$noop" ]; then
+  case "$shell" in
+  fish ) echo "  . (rbenv init -|psub)" ;;
```

Comment:
I'm going to rework this into a single fish/bash switch, and I'll do the
bash one. I'll push the branch to your repo, and if you wouldn't mind,
doing the fish variant.

On Fri, Nov 13, 2015 at 10:52 AM, Max Nordlund <notifications@github.com>
wrote:

> In bin/rbenv-sh-update
> <https://github.com/rkh/rbenv-update/pull/31#discussion_r44797058>:
>
> > +
> > +
> > +
> > +echo "if command rbenv update $@; then"
> > +
> > +if [ -z "$quiet" ]; then
> > +  echo "  if command test -t 1; then"
> > +  echo "    printf \"\\e[1;32mReloading rbenv\\e[0m\\n\";"
> > +  echo "  else"
> > +  echo "    printf \"Reloading rbenv\\n\";"
> > +  echo "  fi;"
> > +fi
> > +
> > +if [ -z "$noop" ]; then
> > +  case "$shell" in
> > +  fish ) echo "  . (rbenv init -|psub)" ;;
>
> This should be source since . has been deprecated, see
> fish-shell/fish-shell#310
> <https://github.com/fish-shell/fish-shell/issues/310> and
> http://fishshell.com/release_notes.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rkh/rbenv-update/pull/31/files#r44797058>.
>